specification/compute/resource-manager/Microsoft.Compute/stable/2021-07-01/compute.json

```
@@ -0,0 +1,15112 @@
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/VirtualMachineScaleSetVMExtension"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get VirtualMachineScaleSet VM extension.": {
+            "$ref": "./examples/compute/GetVirtualMachineScaleSetVMExtensions.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachineScaleSets/{vmScaleSetName}/virtualMachines/{instanceId}/extensions": {
+      "get": {
+        "tags": [
+          "VirtualMachineScaleSetVMExtensions"
+        ],
+        "operationId": "VirtualMachineScaleSetVMExtensions_List",
+        "description": "The operation to get all extensions of an instance in Virtual Machine Scaleset.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "vmScaleSetName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the VM scale set."
+          },
+          {
+            "name": "instanceId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The instance ID of the virtual machine."
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The expand expression to apply on the operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/VirtualMachineScaleSetVMExtensionsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List extensions in Vmss instance.": {
+            "$ref": "./examples/compute/ListVirtualMachineScaleSetVMExtensions.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachineScaleSets/{vmScaleSetName}/virtualmachines/{instanceId}/reimage": {
```

We have some casing inconsistencies (and some other following paths). Is it possible that we fix this here?
```suggestion
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachineScaleSets/{vmScaleSetName}/virtualMachines/{instanceId}/reimage": {
```

---

this change is giving LinfDiff, breakingChange and APIDocReview errors. Reverting.

---

I think if we were to fix it, we should fix all of them.

---

but they will need breaking change exceptions. how does this impact DotNet SDK generation ?

---

Yes, this impacts the DotNet SDK generation because new track 2 of DotNet relies on the hierarchy of the API paths - this casing issue breaks it.